example.py

```diff
@@ -198,6 +209,221 @@ def print_rpc_device(device: RpcDevice) -> None:
+        "device_name": "Test Name",
+        "switch_name": "Switch Test Name",
+        "input_name": "Input Test Name",
+        "script_name": "Script Test Name",
+        "em_name": "Energy Monitor Test Name",
+        "mqtt_server": "mqtt.test.server",
+        "sntp_server": "sntp.test.server",
+        "coiot_peer": "home-assistant.server:8123",
+    }
+
+
+def _normalize_block_data(data: dict[str, Any]) -> dict[str, Any]:
+    """Normalize data for BLOCK devices."""
+    status = data["status"]
+    shelly = data["shelly"]
+    settings = data["settings"]
+    values = _replacement_values()
+
+    real_mac: str = status["mac"]
+    short_mac: str = status["mac"][6:12]
+
+    # Shelly endpoint
+    shelly["name"] = values["device_name"]
+    shelly["mac"] = values["device_mac"]
+
+    # Status endpoint
+    status["mac"] = values["device_mac"]
+
+    if status["wifi_sta"].get("ssid"):
+        status["wifi_sta"]["ssid"] = values["wifi"]
+
+    # Config endpoint
+
+    if settings["coiot"].get("peer"):
+        settings["coiot"]["peer"] = values["coiot_peer"]
+
+    # Some devices use short MAC (uppercase/lowercase)
+    settings["device"]["hostname"] = settings["device"]["hostname"].replace(
+        real_mac, values["device_mac"]
+    )
+    settings["device"]["hostname"] = settings["device"]["hostname"].replace(
+        short_mac, values["device_short_mac"]
+    )
+    settings["device"]["mac"] = values["device_mac"]
+
+    # Some devices use MAC and short MAC (uppercase/lowercase)
+    settings["mqtt"]["id"] = settings["mqtt"]["id"].replace(
+        real_mac, values["device_mac"]
+    )
+    settings["mqtt"]["id"] = settings["mqtt"]["id"].replace(
+        real_mac.lower(), values["device_mac"].lower()
+    )
+    settings["mqtt"]["id"] = settings["mqtt"]["id"].replace(
+        short_mac, values["device_short_mac"]
+    )
+    settings["mqtt"]["id"] = settings["mqtt"]["id"].replace(
+        short_mac.lower(), values["device_short_mac"].lower()
+    )
+
+    settings["name"] = values["device_name"]
+    # Some devices use MAC and short MAC (uppercase/lowercase)
+    settings["wifi_ap"]["ssid"] = settings["wifi_ap"]["ssid"].replace(
+        real_mac, values["device_mac"]
+    )
+    settings["wifi_ap"]["ssid"] = settings["wifi_ap"]["ssid"].replace(
+        short_mac, values["device_short_mac"]
+    )
+    settings["wifi_sta"]["ssid"] = values["wifi"]
+
+    # Updata dataset
+    data.update({"settings": settings})
+    data.update({"shelly": shelly})
+    data.update({"status": status})
+
+    return data
+
+
+def _normalize_rpc_data(data: dict[str, Any]) -> dict[str, Any]:
+    """Normalize data for RPC devices."""
+    config = data["config"]
+    status = data["status"]
+    shelly = data["shelly"]
+
+    real_mac: str = status["sys"]["mac"]
+
+    values = _replacement_values()
+
+    # Config endpoint
+    config["sys"]["device"]["name"] = values["device_name"]
+    config["sys"]["device"]["mac"] = values["device_mac"]
```

Comment:
```suggestion
    device = config["sys"]["device"]
    device["name"] = values["device_name"]
    device["mac"] = values["device_mac"]
```

You can avoid all the dict lookups by getting the `device` dict and than updating it

Reply:
But if I don't change the `config` dict itself, when I write it later I get the original data and not the mutated one.
Am I missing something ?